addons/default/visiosoft/media-field_type/resources/js/upload.js

```diff
@@ -67,23 +68,28 @@ $(function () {
     dropzone.on('success', function (file) {
 
         var response = JSON.parse(file.xhr.response);
-
-        uploaded.push(response.id);
-
-        $('.media-selected-wrapper').load(
-            REQUEST_ROOT_PATH + '/streams/media-field_type/selected?uploaded=' + uploaded.join(','),
-            function () {
-                $('input[name="files"]').val(uploaded.join(','))
-            }
-        );
-
-        file.previewElement.querySelector('[data-dz-uploadprogress]').setAttribute('class', 'progress progress-success');
-
-        setTimeout(function () {
-
-            addAppendByData(uploaded[0])
-            file.previewElement.remove();
-        }, 500);
+        var mimeType = response.mime_type.split('/')
+        if (mimeType[0] === 'image'){
+            uploaded.push(response.id);
+
+            $('.media-selected-wrapper').load(
+                REQUEST_ROOT_PATH + '/streams/media-field_type/selected?uploaded=' + uploaded.join(','),
+                function () {
+                    $('input[name="files"]').val(uploaded.join(','))
+                }
+            );
+
+            file.previewElement.querySelector('[data-dz-uploadprogress]').setAttribute('class', 'progress progress-success');
+
+            setTimeout(function () {
```

Comment:
Instead of adding Timeout, We should the above load incase fails.

Reply:
That code used to be so I didn't change it @ozcan-durak